Add support for widescreen RAW Screen format

andrikpowell opened the pull request:
This is currently a draft.
The main issue I'm having is that there are some garbage pixels on the left side of the screen.

New "Heretic + Hexen" IWADs use widescreen RAW format graphics.
I'd like to support these, as well as any PWADs in the future that use the same widescreen RAW format.

andrikpowell committed:
Add support for widescreen RAW Screen format

rfomin commented:
> The main issue I'm having is that there are some garbage pixels on the left side of the screen.

What resolution? OpenGL or Software? I can't see it.

andrikpowell commented:
> > The main issue I'm having is that there are some garbage pixels on the left side of the screen.
> 
> What resolution? OpenGL or Software? I can't see it.

OpenGL works fine. It just happens in software.

I have a 2560x1440 screen, but I've tested a bunch of other lower resolutions like 1920x1080, 1920x1440, and most lower.
Weirdly enough the weird line of pixels doesn't show for me at 1366x768, 1120x400, 854x400, 560x200, 426x200, 384x200, and 320x200.
Keep in mind this is when "fullscreen" is enabled.

The easiest way to see the issue is by changing "display > status bar > appearance" to "fit to width" and it'll show the garbage pixels pretty much all the time.

Pedro-Beirao commented:
Can confirm this is an issue.
Software, windowed, 768x400, appearance "doom format"

![Screenshot 2025-08-09 at 17 30 08](https://github.com/user-attachments/assets/062e3634-f176-407c-b128-c41c3294c206)

andrikpowell committed:
Fix left screen garbage pixels

andrikpowell committed:
Fix Heretic E3 DemonScroll

andrikpowell commented:
Fixed garbage pixels. Also fixed Heretic E3 scrolling endscreen. Note that the black bars behave weird here, but it's been a known issue (see comment next to `V_ClearBorder()`)

Pedro-Beirao reviewed (approved):
Looks good, good job!
